src/modules/shared/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,66 @@
+import { Sidebar, Menu, MenuItem } from "react-pro-sidebar";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+import ProjectsIcon from "../../../../assets/Components/Projects-icon/ProjectsIcon.jsx";
+
+export default function SideBar() {
+  const [isCollapse, setIsCollapse] = useState<boolean>(
+    localStorage.getItem("sidebar-collapsed") === "true"
```

Comment:
This is okay, but it will execute in each render. A better alternative to use the initialization function syntax which will get evaluate only in the first render.